packages/atomic/src/components/insight/atomic-insight-facet/atomic-insight-facet.tsx

```diff
@@ -0,0 +1,277 @@
+   *
+   * The following facet only uses the `Contact`, `Account`, and `File` values of the `objecttype` field. Even if the
+   * current result set contains other `objecttype` values, such as `Message`, or `Product`, the facet does not use
+   * those other values.
+   *
+   * ```html
+   * <atomic-insight-facet field="objecttype" allowed-values="Contact,Account,File"></div>
+   * ```
+   *
+   * The maximum amount of allowed values is 25.
+   *
+   * Default value is `undefined`, and the facet uses all available values for its `field` in the current result set.
+   */
+  @Prop() public allowedValues?: string;
+
+  @FocusTarget()
+  private showLessFocus!: FocusTargetController;
+
+  @FocusTarget()
+  private showMoreFocus!: FocusTargetController;
+
+  @FocusTarget()
+  private headerFocus!: FocusTargetController;
+
+  public initialize() {
+    const options: InsightFacetOptions = {
+      facetId: this.facetId,
+      field: this.field,
+      numberOfValues: this.numberOfValues,
+      sortCriteria: this.sortCriteria,
+      facetSearch: {numberOfValues: this.numberOfValues},
+      filterFacetCount: this.filterFacetCount,
+      injectionDepth: this.injectionDepth,
+      allowedValues: this.allowedValues?.trim().split(','),
+    };
+
+    this.facet = buildInsightFacet(this.bindings.engine, {options});
+    this.facetId = this.facet.state.facetId;
+
+    this.facetCommon = new FacetCommon({
+      host: this.host,
+      bindings: this.bindings,
+      label: this.label,
+      field: this.field,
+      displayValuesAs: this.displayValuesAs,
+      dependsOn: this.dependsOn,
+      dependenciesManager: buildInsightFacetConditionsManager(
+        this.bindings.engine,
+        {
+          facetId: this.facetId!,
+          conditions: parseDependsOn(this.dependsOn),
+        }
+      ),
+      facet: this.facet,
+      facetId: this.facetId,
+      withSearch: this.withSearch,
+      sortCriteria: this.sortCriteria,
+    });
+
+    this.searchStatus = buildInsightSearchStatus(this.bindings.engine);
+  }
+
+  public disconnectedCallback() {
+    this.facetCommon.disconnectedCallback();
+  }
+
+  public componentShouldUpdate(
+    next: unknown,
+    prev: unknown,
+    propName: keyof AtomicInsightFacet
+  ) {
+    return (
+      !this.facetCommon ||
+      this.facetCommon?.componentShouldUpdate(
+        (next as InsightFacetState).facetSearch,
+        (prev as InsightFacetState).facetSearch,
+        propName
+      )
+    );
+  }
+
+  public render() {
+    if (!this.facetCommon) {
+      return (
+        <FacetPlaceholder
+          numberOfValues={this.numberOfValues}
+          isCollapsed={this.isCollapsed}
+        ></FacetPlaceholder>
+      );
+    }
+    return this.facetCommon.render({
+      hasError: this.searchStatusState.hasError,
+      firstSearchExecuted: this.searchStatusState.firstSearchExecuted,
+      isCollapsed: this.isCollapsed,
+      numberOfValues: this.numberOfValues,
+      headerFocus: this.headerFocus,
```

Comment:
I'd prefer if values that aren't meant to be changed like *Focus and perhaps even headingLevel were passed at initialization instead.

Reply:
the problem with the focus is that because of the decorator, it happens after initialization, so it needs to go here!
but the hading level can go to initialization, I'll update